### DISCUSSION TITLE
ReferenceError: regeneratorRuntime is not defined

### DISCUSSION THREAD

**Discussion by gogo9th** [babel/babel#13103] [General] [answered]
Hi,

I keep getting `ReferenceError: regeneratorRuntime is not defined`. My goal is to run the transformed code in a `<script>` tag of a browser webpage. I tried to search Google and this repository's Discussion section to find the solution, but could not find the exact way of solving this. The following is the Babel option I am using in the code. Could some guru please fix present the exact syntax for fixing this error? Thanks a lot in advance. 

```
      var options = {
        retainLines: true,
        compact: false,
        sourceType: 'unambiguous',
        sourceMaps: true,
        presets: [ ['@babel/preset-env',
                     { "exclude": ["@babel/plugin-transform-function-name"] }
                  ] ],
        plugins: [
          ["transform-es2015-modules-commonjs-simple",
            {
              'noMangle': true,
              'allowTopLevelThis': true,
              'strictMode': false
            },
          ]
        ]
      }
```

**Comment by nicolo-ribaudo** [comment, marked as answer]
Of you are using `<script>` tags, you can add a new script tag _before_ the one running your code, with `src` set to `https://unpkg.com/regenerator-runtime@0.13.7/runtime.js`

**Reply by gogo9th** [reply]
Thanks so much, this indeed works! Out of curiosity, is there also a solution purely based on setting up the above Babel option properly?

**Reply by nicolo-ribaudo** [reply]
You can use `@babel/plugin-transform-runtime` so that it automatically injects imports to that "helper", but then you need a bundler (webpack/Rollup/other) to resolve and bundle those imports.

**Reply by gogo9th** [reply]
Got it, thanks.
